proto/substrait/algebra.proto

```diff
@@ -478,47 +478,136 @@ message Expression {
     }
   }
 
+  // A scalar function call.
   message ScalarFunction {
-    // points to a function_anchor defined in this plan
+    // Points to a function_anchor defined in this plan, which must refer
+    // to a scalar function in the associated YAML file. Required; 0 is
```

Comment:
I did, but I ended up yielding to just allowing it, especially since type variation 0 is somehow functional, and if we ever need an optional reference we can just stick the int in a message. I either already changed it in the validator or it's waiting in one of the open PRs. I think I did leave in a discouraging note diagnostic, though.

Reply:
Your call. I'm fine with either.

Reply:
I just used "avoid" in the end, since at the very least it's confusing in the JSON serialization (since protobuf will omit fields with default values).